internal/journal.go

```diff
@@ -0,0 +1,89 @@
+package internal
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/wal-g/tracelog"
+
+	"github.com/wal-g/wal-g/pkg/storages/storage"
+)
+
+const (
+	JournalSize = "JournalSize"
+)
+
+func AddJournalSizeToPreviousBackup(
+	root storage.Folder,
+	journalPath string,
+	sentinelPath string,
+	sentinelName string,
+	oldBackupTime time.Time,
+	newBackupTime time.Time,
+) error {
+	if len(sentinelName) == 0 {
+		return fmt.Errorf("sentinel name is empty")
+	}
+
+	folder := root.GetSubFolder(sentinelPath)
+	sentinelReader, err := folder.ReadObject(sentinelName)
+	if err != nil {
+		return err
+	}
+
+	rawSentinel, err := io.ReadAll(sentinelReader)
+	if err != nil {
+		return err
+	}
+
+	var sentinel map[string]interface{}
+	err = json.Unmarshal(rawSentinel, &sentinel)
+	if err != nil {
+		return err
+	}
+
+	journalSize, err := GetJournalSizeInSemiInterval(root, journalPath, oldBackupTime, newBackupTime)
+	if err != nil {
+		return err
+	}
+
+	sentinel[JournalSize] = journalSize
+	sentinelJSON, err := json.Marshal(sentinel)
+	if err != nil {
+		return err
+	}
+
+	r := bytes.NewReader(sentinelJSON)
+	err = folder.PutObject(sentinelName, r)
```

Comment:
I suspect that this will update sentinel's `LastModified` object's timestamp... and then it will break all our logic of sorting / selecting / handling backups.

e.g. getLastUploadedBackupSentinel() will always return one backup, isn't it?

Reply:
Yes, I agree with you. I think we can use similar logic to MongoDB: create a file "backup.json" with information about backup sizes, binlog sizes and etc., and just read/update it instead of sentinels.